R/RecallByManufactuer.R

```diff
@@ -0,0 +1,68 @@
+#ua <- user_agent("http://github.com/WrathSmith/caRecall")
+#ua
+#' get_recall_by_number
+#'
+#' @param manufacturer: A car manufacturer name (ie. Honda, Nissan, etc.)
+#' limit: Number of records to return (default is 25)
+#' page: the serach page result to return (default is 1)
+#'
+#'
+#' @return text output
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' get_recall_by_number(1977044)
+#' }
+recall_manufacturer <- function(manufacturer, limit=25, page=1) {
+
+    # refer to: https://httr.r-lib.org/articles/api-packages.html
+
+    # format the url string
+    url_ <- "https://vrdb-tc-apicast-production.api.canada.ca/eng/vehicle-recall-database/v1/recall/manufacturer-name/"
+    url_ <- paste(url_, toString(manufactuer), sep="")
+
+    # query the api
+    response <- httr::GET(url_, add_headers("user-key"=Sys.getenv("VRD_API"),"limit"=limit, "page"=page))
+
+    #check to verify json response
+    if (http_type(response) != "application/json") {
+        stop("API did not return json", call. = FALSE)
+    }
+
+    #parses response
+    parsed <- jsonlite::fromJSON(content(response, "text"), simplifyVector = FALSE)
+
+    if (status_code(response) != 200) {
+        stop(
+            sprintf(
+                "Vehicle Recall Database request failed [%s]\n%s\n<%s>",
+                status_code(response),
+                parsed$message,
+                parsed$documentation_url
+            ),
+            call. = FALSE
+        )
+    }
+
+    #httr::content(response)
+
+    structure(
+        list(
+            content = parsed,
+            name = manufacturer,
+            response = response
+        ),
+        class = "vrd_api"
+    )
+
+}
```

Comment:
I believe all of the above code should go in a helper function called something like `call_vrd_api()` in the `utils.R` file. This code will be called for every API call I believe. Each user facing function should just make a call to `call_vrd_api()` with the required different parameters passed in (similar to the lab). Everything below `url` should likely go in this helper function. The helper function would then output the vrd_api structure and the user function would extract and output a dataframe from the S3.